generate.py

```diff
@@ -246,6 +246,13 @@ def _load_model(checkpoint_path, device, precision, use_tp):
         apply_tp(model)
 
     model = model.to(device=device, dtype=precision)
+    if "int4" in str(checkpoint_path):
+        from quantize import WeightOnlyInt4Linear
+        for fqn, mod in model.named_modules():
+            if isinstance(mod, WeightOnlyInt4Linear):
+                weight = mod.weight.data
+                weight_int4pack = torch.ops.aten._convert_weight_to_int4pack(weight, mod.inner_k_tiles)
+                mod.weight = weight_int4pack
```

Comment:
Can you put the weight conversion into ```WeightOnlyInt4QuantHandler.convert_for_runtime``` at L236? More concretely, it can be part of ```replace_linear_int4```,  I think that's the right place. And your change doesn't work well with TP, since the conversion happens after L246. Otherwise, this looks good!

Reply:
@yanboliang Thanks for the comments!

When quantizing, it generates [n][k / 2] uint8 (serialized format). After model loading to specific device, int4 pack weight can be converted then. So, moving to `WeightOnlyInt4QuantHandler.convert_for_runtime` or `replace_linear_int4` cannot be a suitable way.

I try to make some refactor to wrapper the conversion into a new function and change TP after model loading and weight convert. Is this okay for you? Thanks!